src/main/java/app/onepass/organizer/services/EventService.java

```diff
@@ -0,0 +1,287 @@
+
+		ServiceUtil.saveEntity(eventMessage, eventRepository);
+
+		Result result = ServiceUtil.returnSuccessful("Event creation successful.");
+
+		ServiceUtil.configureResponseObserver(responseObserver, result);
+	}
+
+	@Override
+	public void getEvent(UserRequest request, StreamObserver<GetEventResponse> responseObserver) {
+
+		List<EventEntity> allEventEntities = eventRepository.findAll();
+
+		List<Event> allEvents = allEventEntities.stream()
+				.map(eventEntity -> eventEntity.parseEntity().getEvent())
+				.collect(Collectors.toList());
+
+		GetEventResponse getEventResponse = GetEventResponse.newBuilder()
+				.addAllEvents(allEvents).build();
+
+		ServiceUtil.configureResponseObserver(responseObserver, getEventResponse);
+	}
+
+	@Override
+	public void getEventById(GetByIdRequest request, StreamObserver<GetEventByIdResponse> responseObserver) {
+
+		EventEntity eventEntity;
+
+		try {
+
+			eventEntity = eventRepository
+					.findById(request.getReadId())
+					.orElseThrow(IllegalArgumentException::new);
+
+		} catch (IllegalArgumentException illegalArgumentException) {
+
+			GetEventByIdResponse result = GetEventByIdResponse
+					.newBuilder()
+					.build();
+
+			ServiceUtil.configureResponseObserver(responseObserver, result);
+
+			return;
+		}
+
+		Event event = eventEntity.parseEntity().getEvent();
+
+		GetEventByIdResponse getEventByIdResponse = GetEventByIdResponse
+				.newBuilder()
+				.setEvent(event)
+				.build();
+
+		ServiceUtil.configureResponseObserver(responseObserver, getEventByIdResponse);
+
+	}
+
+	@Override
+	@Transactional
+	public void updateEventInfo(UpdateEventInfoRequest request, StreamObserver<Result> responseObserver) {
+
+		if (!eventRepository.findById(request.getEvent().getId()).isPresent()) {
+
+			Result result = ServiceUtil.returnError("An event with this ID does not exist.");
+
+			ServiceUtil.configureResponseObserver(responseObserver, result);
+
+			return;
+		}
+
+		EventMessage eventMessage = new EventMessage(request.getEvent());
+
+		ServiceUtil.saveEntity(eventMessage, eventRepository);
+
+		Result result = ServiceUtil.returnSuccessful("Event update successful.");
+
+		ServiceUtil.configureResponseObserver(responseObserver, result);
+	}
+
+	@Override
+	@Transactional
+	public void removeEvent(RemoveEventRequest request, StreamObserver<Result> responseObserver) {
+
+		long eventId = request.getEventId();
+
+		boolean deleteSuccessful = ServiceUtil.deleteEntity(eventId, eventRepository);
+
+		if (!deleteSuccessful) {
+
+			Result result = ServiceUtil.returnError("Cannot find event from given ID.");
+
+			ServiceUtil.configureResponseObserver(responseObserver, result);
+
+			return;
+
+		}
+
+		Result result = ServiceUtil.returnSuccessful("Event deletion successful.");
+
+		ServiceUtil.configureResponseObserver(responseObserver, result);
+	}
+
+	@Override
+	@Transactional
+	public void updateEventFacility(UpdateEventFacilityRequest request, StreamObserver<Result> responseObserver) {
```

Comment:
Why need this since event only can add facility via facility request?

Reply:
for when organizers want to change facility for an event

Reply:
But each facility need to be change though facility request?

Reply:
I can create method for you in facility service